models/image_classification/imagenet_postprocess.py

```diff
@@ -0,0 +1,13 @@
+import mxnet as mx
```

Comment:
`import numpy as np`

Reply:
the only use of `mx` is the softmax function.

You can implement the `softmax` like this

```
def softmax(X, theta = 1.0, axis = None):
    """
    Compute the softmax of each element along an axis of X.

    Parameters
    ----------
    X: ND-Array. Probably should be floats. 
    theta (optional): float parameter, used as a multiplier
        prior to exponentiation. Default = 1.0
    axis (optional): axis to compute values along. Default is the 
        first non-singleton axis.

    Returns an array the same size as X. The result will sum to 1
    along the specified axis.
    """

    # make X at least 2d
    y = np.atleast_2d(X)

    # find axis
    if axis is None:
        axis = next(j[0] for j in enumerate(y.shape) if j[1] > 1)

    # multiply y against the theta parameter, 
    y = y * float(theta)

    # subtract the max for numerical stability
    y = y - np.expand_dims(np.max(y, axis = axis), axis)
    
    # exponentiate y
    y = np.exp(y)

    # take the sum along the specified axis
    ax_sum = np.expand_dims(np.sum(y, axis = axis), axis)

    # finally: divide elementwise
    p = y / ax_sum

    # flatten if X was 1D
    if len(X.shape) == 1: p = p.flatten()

    return p
```

reference: https://nolanbconaway.github.io/blog/2017/softmax-numpy

Reply:
@houseroad we all know what softmax is, and how to implement it.
However, I think using a library function makes a better case for readability and reducing the cognitive load on users, so I don't think it makes sense to implement softmax here.
As a side note, the issue of using framework specific code (in this case MXNet) for pre/post processing in a separate file was already discussed and approved in a review meeting.

Reply:
Yes, people know `softmax`. My point is that if we can get rid of the dependency of specific backend, user experience can be improved a lot. Not all the users would like to install another backend just for preprocessing and postprocessing. If they don't install mxnet, they have write the preprocessing and postprocessing script by themselves. Such changes for us are so easy to make, that's why I suggest to create simple libraries using numpy.

Reply:
@lupesko - I think this goes back to our discussion around neutrality and using OpenCV/Numpy for generic pre/post processing. I think it is OK to link to mxnet examples but the model zoo, as we discussed, should not be any more tied to a particular platform than required. Sound reasonable?

Reply:
The agreement was to not have framework specific code in the model readme, and instead link framework specific file(s). I think this is exactly what the contributor has done here.

Looks like you are asking to host this python code outside of the repo? if we do that, the user experience will be exactly the same (link to a code file), but maintenance and review will be more difficult (since we will have cross repos dependencies). Please consider this point...